libs/Node.py

```diff
@@ -38,10 +40,14 @@ def __init__(self,
         # setup network and routing related controllers
         self.network_controller = network_controller
         self.neighbours_controller = NeighboursController(
-            config_controller=self.config_controller, 
+            config_controller=self.config_controller,
             network=network_controller
         )
-        
+        self.routing_controller = RoutingController(
+            self.neighbours_controller, network_controller)
+
+        # TODO: Fix this with a different method
+        self.network_controller.routing_controller = self.routing_controller  # type: ignore
```

Comment:
wouldn't it be easier to create the routing controller in the E220NetworkController

Reply:
Yes and no.

Whilst it would be ever so slightly easier to, it would also spread out the code more. Now we still have one place where we define all the callbacks for the network controller, and one place where we start asyncio tasks. I think having that central is more important than that tiny improvement.